worker/provisioner/broker.go

```diff
@@ -98,6 +98,13 @@ func finishNetworkConfig(bridgeDevice string, interfaces []network.InterfaceInfo
 	}
 
 	if !haveNameservers || !haveSearchDomains {
+		if !haveNameservers {
```

Comment:
Might be worth accruing a []string with "name servers" and/or "search domains", then having the single log message like:

```logger.Warningf("DNS config is missing %s, discovering host's DNS config", strings.split(missing, " and ")```